app/models/concerns/hyku_addons/bulkrax/object_factory_behavior.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+module HykuAddons
+  module Bulkrax
+    module ObjectFactoryBehavior
+      def run
+        arg_hash = { id: attributes[:id], name: "UPDATE", klass: klass }
+        @object = find
+
+        if object
+          object.reindex_extent = Hyrax::Adapters::NestingIndexAdapter::LIMITED_REINDEX if object.respond_to? :reindex_extent
+          ActiveSupport::Notifications.instrument("import.importer", arg_hash) { update }
+        else
+          ActiveSupport::Notifications.instrument("import.importer", arg_hash.merge(name: "CREATE")) { create }
+        end
+
+        yield(object) if block_given?
+
+        object
+      end
+
+      # An ActiveFedora bug when there are many habtm <-> has_many associations means they won't all get saved.
+      # https://github.com/projecthydra/active_fedora/issues/874
+      # 2+ years later, still open!
+      # rubocop:disable Metrics/MethodLength
+      def create
+        attrs = create_attributes
+        @object = klass.new
+        object.reindex_extent = Hyrax::Adapters::NestingIndexAdapter::LIMITED_REINDEX if object.respond_to? :reindex_extent
+        run_callbacks :save do
+          run_callbacks :create do
+            if klass == ::AdminSet
+              create_admin_set(attrs)
+            elsif klass == ::Collection
+              create_collection(attrs)
+            else
+              work_actor.create(environment(attrs))
+            end
+          end
+        end
+        log_created(object)
+      end
+      # rubocop:enable Metrics/MethodLength
+
+      # rubocop:disable Metrics/PerceivedComplexity
+      # rubocop:disable Metrics/CyclomaticComplexity
+      def update
+        raise "Object doesn't exist" unless object
+
+        destroy_existing_files if @replace_files && (klass != ::Collection || klass != ::AdminSet)
+        attrs = update_attributes
+        run_callbacks :save do
+          if klass == ::AdminSet
```

Comment:
This bit is common to both update and create so could be moved into another method to reduce the complexity.